internal/common/metrics.go

```diff
@@ -132,3 +155,120 @@ func checkMetricPresent(mcClient *client.Client, namespace string, podName strin
 		return nil
 	}
 }
+
+func runTestPod(mcClient *client.Client, podName string, ns string) error {
+	t := true
+	f := false
+	userAndGroup := int64(35)
+
+	pod := corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      podName,
+			Namespace: ns,
+		},
+		Spec: corev1.PodSpec{
+			SecurityContext: &corev1.PodSecurityContext{
+				RunAsUser:    &userAndGroup,
+				RunAsGroup:   &userAndGroup,
+				RunAsNonRoot: &t,
+				SeccompProfile: &corev1.SeccompProfile{
+					Type: "RuntimeDefault",
+				},
+			},
+			Containers: []corev1.Container{
+				{
+					Name:  "test",
+					Image: "alpine:latest",
+					Args:  []string{"sleep", "99999999"},
+					SecurityContext: &corev1.SecurityContext{
+						Capabilities: &corev1.Capabilities{
+							Drop: []corev1.Capability{
+								"ALL",
+							},
+						},
+						AllowPrivilegeEscalation: &f,
+					},
+				},
+			},
+		},
+	}
+	// Check if pods exists already.
+	create := false
+	existing := corev1.Pod{}
+	err := mcClient.Get(context.Background(), client2.ObjectKey{Namespace: ns, Name: podName}, &existing)
+	if errors.IsNotFound(err) {
+		create = true
+	} else if err != nil {
+		return fmt.Errorf("error ensuring test pod is deleted %s: %s", podName, err)
+	}
+
+	if !create {
+		// Check if pod is running.
+		if existing.Status.Phase != corev1.PodRunning {
+			// Pod unhealthy, delete and recreate it.
+			err := cleanupTestPod(mcClient, podName, ns)
+			if err != nil {
+				return err
+			}
+
+			create = true
+		}
+	}
+
+	if create {
+		// Create the pod.
+		err = mcClient.Create(context.Background(), &pod)
+		if err != nil {
+			return fmt.Errorf("can't create test pod %s: %s", podName, err)
+		}
+	}
+
+	// Wait for pod to be running.
+	for {
+		err = mcClient.Get(context.Background(), client2.ObjectKey{Namespace: ns, Name: podName}, &existing)
+		if err != nil {
+			return fmt.Errorf("error ensuring test pod is running %s: %s", podName, err)
+		}
+
+		if existing.Status.Phase == corev1.PodRunning {
+			break
+		}
+
+		logger.Log("Waiting for pod %s to be running", podName)
+		time.Sleep(5 * time.Second)
+	}
```

Comment:
Instead of using these endless `for` loops could we instead please use something like `Eventually` with a timeout so that we don't wait forever if something gets stuck?

Something like... (I haven't tested this)

```
Eventually(func() (bool, error) {
	err = mcClient.Get(context.Background(), client2.ObjectKey{Namespace: ns, Name: podName}, &existing)
	if err != nil {
		return false, fmt.Errorf("error ensuring test pod is running %s: %s", podName, err)
	}

	if existing.Status.Phase == corev1.PodRunning {
		return true, nil
	}

	logger.Log("Waiting for pod %s to be running", podName)
	return false, nil
}).
	WithTimeout(5 * time.Minute).
	WithPolling(5 * time.Second).
	Should(BeTrue())
```


Same with below when deleting.

Reply:
done, testing in progress

Reply:
seems to work